analyzer/vulnerability-consumer/src/main/java/eu/fasten/analyzer/vulnerabilityconsumer/db/MetadataUtility.java

```diff
@@ -0,0 +1,375 @@
+                .from(Callables.CALLABLES)
+                .join(Modules.MODULES).on(Callables.CALLABLES.MODULE_ID.eq(Modules.MODULES.ID))
+                .join(PackageVersions.PACKAGE_VERSIONS).on(Modules.MODULES.PACKAGE_VERSION_ID.eq(PackageVersions.PACKAGE_VERSIONS.ID))
+                .join(Packages.PACKAGES).on(PackageVersions.PACKAGE_VERSIONS.PACKAGE_ID.eq(Packages.PACKAGES.ID))
+                .where(Callables.CALLABLES.ID.eq(id))
+                .fetchOne();
+
+        var partialFastenUri = callInfo.component1();
+        var namespace = callInfo.component2();
+        var pkgVersion = callInfo.component3();
+        var pkgName = callInfo.component4();
+
+        return generateFullFastenUri(pkgName, pkgVersion, namespace, partialFastenUri);
+    }
+
+    //////////////////////////
+    //    DATABASE INJECT   //
+    //////////////////////////
+
+    /**
+     * Injects information at package level.
+     *
+     * @param v            - Vulnerability Object
+     * @param pkgVersionId - ID of the package version
+     * @param context      - DSLContext to use to inject the data
+     */
+    public void injectPackageVersionVulnerability(Vulnerability v, Long pkgVersionId, DSLContext context) {
+        logger.info("Injecting vulnerability " + v.getId() + " into PACKAGE VERSION with ID: " + pkgVersionId);
+        // Step 1: Get metadata JSONObject of the pkgVersionId
+        var metadataRecord = context.select(PackageVersions.PACKAGE_VERSIONS.METADATA)
+                .from(PackageVersions.PACKAGE_VERSIONS)
+                .where(PackageVersions.PACKAGE_VERSIONS.ID.equal(pkgVersionId))
+                .fetchOne()
+                .component1();
+
+        var metadata = new JSONObject(metadataRecord.data());
+
+        // Step 2: Check if vulnerabilities exist, else create a map
+        if (!metadata.has("vulnerabilities"))
+            metadata.put("vulnerabilities", new JSONObject());
+
+        // Step 3: Get the JSON of the vulnerability and put it in the metadata
+        try {
+            var vulnJson = objectMapper.writeValueAsString(v);
+            ((JSONObject) metadata.get("vulnerabilities")).put(v.getId(), new JSONObject(vulnJson));
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
+        }
+
+        // Step 4: Update the value of the metadata in the DB
+        context.update(PackageVersions.PACKAGE_VERSIONS)
+                .set(PackageVersions.PACKAGE_VERSIONS.METADATA, JSONB.valueOf(metadata.toString()))
+                .where(PackageVersions.PACKAGE_VERSIONS.ID.equal(pkgVersionId))
+                .execute();
+    }
+
+    /**
+     * Injects information at callable level.
+     *
+     * @param v          - Vulnerability Object
+     * @param callableId - ID of the callable
+     * @param context    - DSLContext to use to inject the data
+     * @return
+     */
+    public int injectCallableVulnerability(Vulnerability v, Long callableId, DSLContext context) {
```

Comment:
The same goes for this one (https://github.com/fasten-project/fasten/blob/develop/core/src/main/java/eu/fasten/core/data/metadatadb/MetadataDao.java#L635)